### PR TITLE
Populates existing Battles new fields with existing Battle Teams data

### PR DESCRIPTION
**Pull request by jaenia**
[As the opponent Trainer I select my PKNs to battle](https://trello.com/c/oXVCG30U/4-as-the-opponent-trainer-i-select-my-pkns-to-battle)

### Description
This PR adds migration to update new fields (creator pokemons) of `Battle` model with existing data in `BattleTeam`.

### How to test
You need to have at least one `BattleTeam` and one related `Battle` created in your database.
- [ ]  Run `python manage.py migrate`
- [ ] Check if the new fields of your battle is updated according to battle team information
- [ ] Check that you can revert the operation

### CODE REVIEW

File: backend/battles/migrations/0007_auto_20210113_1745.py [hunk 1]
```diff
@@ -0,0 +1,46 @@
+# Generated by Django 2.2.17 on 2021-01-13 17:45
+
+from django.db import migrations
+
+
+def forwards_func(apps, schema_editor):
+    Battle = apps.get_model("battles", "Battle")
+    BattleTeam = apps.get_model("battles", "BattleTeam")
+
+    db_alias = schema_editor.connection.alias
```
Comment (filipeximenes) [inline backend/battles/migrations/0007_auto_20210113_1745.py]: Why do we need this?

File: backend/battles/migrations/0007_auto_20210113_1745.py [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+        Battle.objects.using(db_alias).filter(
+            id=battle_team.battle.id,
+        ).update(
+            creator_pokemon_1=battle_team.pokemon_1,
+            creator_pokemon_2=battle_team.pokemon_2,
+            creator_pokemon_3=battle_team.pokemon_3,
+        )
+
+
+def reverse_func(apps, schema_editor):
```
Comment (filipeximenes) [inline backend/battles/migrations/0007_auto_20210113_1745.py:20]: It's ok to have this here, but usually in this situation we wouldn't write a reverse function

File: backend/battles/migrations/0007_auto_20210113_1745.py (hunk 1 above)
Comment (jaenia) [reply backend/battles/migrations/0007_auto_20210113_1745.py]: Oh, sorry. That is used in this Django documentation example: https://docs.djangoproject.com/en/3.1/ref/migration-operations/#django.db.migrations.operations.RunPython
But that is helpful when we have multiple databases, which is not the case here. I'll remove it.

File: backend/battles/migrations/0007_auto_20210113_1745.py (hunk 2 above)
Comment (jaenia) [reply backend/battles/migrations/0007_auto_20210113_1745.py:20]: Hmm, interesting. Could you please tell me why? I thought that was important in the case we need to back for a previous version of the database.

Comment (filipeximenes) [reply backend/battles/migrations/0007_auto_20210113_1745.py:20]: For the context on this particular use case in this application there's no gain in setting this fields as null. you can always go back and forth with this migration and the result is the same. Eg.: suppose there's a bug in `forwards_func`. You could just revert the migration (doing nothing), fix `forwards_func` and then run the migration again. 

You will need to evaluate case by case if you need or not a migration reversion function

**Review by filipeximenes — APPROVED** [review]

File: backend/battles/migrations/0007_auto_20210113_1745.py (hunk 2 above)
Comment (jaenia) [reply backend/battles/migrations/0007_auto_20210113_1745.py:20]: Got it!

**Pull request merged**